Add logging and Web Inspector support for spatial and projection metadata

jernoble opened the pull request:
#### 71b6d3a51a9e0df507c3ddb7cb63afe1914a709d
<pre>
Add logging and Web Inspector support for spatial and projection metadata
<a href="https://rdar.apple.com/150615061">rdar://150615061</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=292507">https://bugs.webkit.org/show_bug.cgi?id=292507</a>

Reviewed by Jean-Yves Avenard.

Add an enumerated list of projection types, and pass that metadata as well
as spatialVideoMetadata through to the Web Inspector, as well as the UIProcess.

Drive-by fixes:
- Add a change observer to VideoTrackConfiguration to allow clients to be notified
  of changes to the configuration without requiring a Client superclass.
- Make AVTrackPrivateAVFObjCImpl reference-counted.
- Compare AVAssetTrack.mediaType and AVMediaSelectionOption.mediaType with -isEqualToString:
  rather than pointer equality.

* Source/JavaScriptCore/inspector/protocol/DOM.json:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Logger.h:
(WTF::LogArgument&lt;std::optional&lt;T&gt;&gt;::toString):
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
(WTF::LogArgument&lt;std::optional&lt;T&gt;&gt;::toString): Deleted.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::audioTrackConfigurationChanged):
(WebCore::HTMLMediaElement::videoTrackConfigurationChanged):
(WebCore::HTMLMediaElement::addAudioTrack):
(WebCore::HTMLMediaElement::addVideoTrack):
(WebCore::HTMLMediaElement::removeAudioTrack):
(WebCore::HTMLMediaElement::removeVideoTrack):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::descriptionForTrack):
(WebCore::MediaElementSession::description const):
* Source/WebCore/html/MediaElementSession.h:
* Source/WebCore/html/track/AudioTrack.cpp:
(WebCore::AudioTrack::configurationChanged):
* Source/WebCore/html/track/AudioTrackClient.h:
(WebCore::AudioTrackClient::audioTrackConfigurationChanged):
* Source/WebCore/html/track/VideoTrack.cpp:
(WebCore::VideoTrack::configurationChanged):
* Source/WebCore/html/track/VideoTrackClient.h:
(WebCore::VideoTrackClient::videoTrackConfigurationChanged):
* Source/WebCore/html/track/VideoTrackConfiguration.cpp:
(WebCore::VideoTrackConfiguration::setState):
(WebCore::VideoTrackConfiguration::setCodec):
(WebCore::VideoTrackConfiguration::setWidth):
(WebCore::VideoTrackConfiguration::setHeight):
(WebCore::VideoTrackConfiguration::setColorSpace):
(WebCore::VideoTrackConfiguration::setFramerate):
(WebCore::VideoTrackConfiguration::setBitrate):
(WebCore::VideoTrackConfiguration::setSpatialVideoMetadata):
(WebCore::VideoTrackConfiguration::setVideoProjectionMetadata):
(WebCore::VideoTrackConfiguration::notifyObservers):
(WebCore::VideoTrackConfiguration::toJSON const):
* Source/WebCore/html/track/VideoTrackConfiguration.h:
(WebCore::VideoTrackConfiguration::videoProjectionMetadata const):
(WebCore::VideoTrackConfiguration::setState): Deleted.
(WebCore::VideoTrackConfiguration::setCodec): Deleted.
(WebCore::VideoTrackConfiguration::setWidth): Deleted.
(WebCore::VideoTrackConfiguration::setHeight): Deleted.
(WebCore::VideoTrackConfiguration::setColorSpace): Deleted.
(WebCore::VideoTrackConfiguration::setFramerate): Deleted.
(WebCore::VideoTrackConfiguration::setBitrate): Deleted.
(WebCore::VideoTrackConfiguration::setSpatialVideoMetadata): Deleted.
(WebCore::VideoTrackConfiguration::isImmersiveVideo const): Deleted.
(WebCore::VideoTrackConfiguration::setIsImmersiveVideo): Deleted.
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::videoProjectionMetadataKind):
(WebCore::InspectorDOMAgent::getMediaStats):
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
(WebCore::PlaybackSessionModelClient::supportsLinearMediaPlayerChanged):
(WebCore::PlaybackSessionModelClient::spatialVideoMetadataChanged):
(WebCore::PlaybackSessionModelClient::videoProjectionMetadataChanged):
(WebCore::PlaybackSessionModelClient::isImmersiveVideoChanged): Deleted.
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
(WebCore::PlaybackSessionModelMediaElement::videoTrackConfigurationChanged):
(WebCore::PlaybackSessionModelMediaElement::maybeUpdateVideoMetadata):
(WebCore::PlaybackSessionModelMediaElement::seekableRanges const):
(WebCore::PlaybackSessionModelMediaElement::isInWindowFullscreenActive const):
(WebCore::PlaybackSessionModelMediaElement::logIdentifier const):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::convertEnumerationToString):
(WebCore::convertVideoProjectionMetadataToString):
* Source/WebCore/platform/graphics/PlatformVideoTrackConfiguration.h:
* Source/WebCore/platform/graphics/VideoProjectionMetadata.h: Copied from Source/WebCore/html/track/VideoTrackClient.h.
(WTF::LogArgument&lt;WebCore::VideoProjectionMetadata&gt;::toString):
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.mm:
(WebCore::AVTrackPrivateAVFObjCImpl::initializeAssetTrack):
(WebCore::AVTrackPrivateAVFObjCImpl::initializationCompleted):
(WebCore::AVTrackPrivateAVFObjCImpl::readyState const):
(WebCore::AVTrackPrivateAVFObjCImpl::videoTrackConfiguration const):
(WebCore::AVTrackPrivateAVFObjCImpl::spatialVideoMetadata const):
(WebCore::AVTrackPrivateAVFObjCImpl::videoProjectionMetadata const):
(WebCore::AVTrackPrivateAVFObjCImpl::isImmersiveVideo const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp:
(WebCore::spatialVideoMetadataFromFormatDescription):
(WebCore::videoProjectionMetadataFromFormatDescription):
(WebCore::videoMetadataFromFormatDescription): Deleted.
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.h:
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateAVFObjC.mm:
(WebCore::AudioTrackPrivateAVFObjC::AudioTrackPrivateAVFObjC):
(WebCore::AudioTrackPrivateAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateMediaSourceAVFObjC.cpp:
(WebCore::AudioTrackPrivateMediaSourceAVFObjC::AudioTrackPrivateMediaSourceAVFObjC):
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/TextTrackPrivateMediaSourceAVFObjC.cpp:
(WebCore::m_impl):
* Source/WebCore/platform/graphics/avfoundation/objc/TextTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateAVFObjC.cpp:
(WebCore::VideoTrackPrivateAVFObjC::VideoTrackPrivateAVFObjC):
(WebCore::VideoTrackPrivateAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.mm:
(WebCore::VideoTrackPrivateMediaSourceAVFObjC::VideoTrackPrivateMediaSourceAVFObjC):
* Source/WebCore/platform/graphics/cocoa/NullVideoPresentationInterface.h:
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.cpp:
(WebCore::VideoTrackPrivateWebM::updateConfiguration):
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js:
(WI.MediaDetailsSidebarPanel):
(WI.MediaDetailsSidebarPanel.prototype.layout):
(WI.MediaDetailsSidebarPanel.prototype.initialLayout):
(WI.MediaDetailsSidebarPanel.prototype.localizedVideoProjectionMetadataKindString):
(WI.MediaDetailsSidebarPanel.prototype.setVideo):
* Source/WebKit/Platform/IPC/ArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;JSON::Value&gt;::encode):
(IPC::ArgumentCoder&lt;JSON::Value&gt;::decode):
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.h:
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.mm:
(WebKit::PlaybackSessionInterfaceLMK::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionInterfaceLMK::isImmersiveVideoChanged): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionModelContext::spatialVideoMetadataChanged):
(WebKit::PlaybackSessionModelContext::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionManagerProxy::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionModelContext::isImmersiveVideoChanged): Deleted.
(WebKit::PlaybackSessionManagerProxy::isImmersiveVideoChanged): Deleted.
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionInterfaceContext::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionManager::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionInterfaceContext::isImmersiveVideoChanged): Deleted.
(WebKit::PlaybackSessionManager::isImmersiveVideoChanged): Deleted.

Canonical link: <a href="https://commits.webkit.org/294733@main">https://commits.webkit.org/294733@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b4df94f916b75c47afa684f4408ac4392a5f6fb5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102882 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22556 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12876 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108048 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/53523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104921 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22876 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31057 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/78225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/53523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105888 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17716 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/92842 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/58559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/17557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10878 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52880 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/95557 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87366 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10945 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110423 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/101492 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30019 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/22113 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/87207 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30383 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89038 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/86826 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22100 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31674 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9388 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/24281 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29946 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/35268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/125125 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/29754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34719 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33081 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/31316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateAVFObjC.h:
```diff
@@ -75,11 +75,11 @@ class VideoTrackPrivateAVFObjC final : public VideoTrackPrivateAVF {
     void resetPropertiesFromTrack();
     void videoTrackConfigurationChanged();
-    std::unique_ptr<AVTrackPrivateAVFObjCImpl> m_impl;
+    Ref<AVTrackPrivateAVFObjCImpl> m_impl;
```
```suggestion
    const Ref<AVTrackPrivateAVFObjCImpl> m_impl;
```

will remove the CPP failures too.

jyavenard commented on Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.h:
```diff
@@ -60,7 +60,7 @@ class VideoTrackPrivateMediaSourceAVFObjC final : public VideoTrackPrivateAVF {
     void resetPropertiesFromTrack();
 
-    std::unique_ptr<AVTrackPrivateAVFObjCImpl> m_impl;
+    Ref<AVTrackPrivateAVFObjCImpl> m_impl;
```
```suggestion
    const Ref<AVTrackPrivateAVFObjCImpl> m_impl;
```

jyavenard commented on Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
```diff
@@ -1302,7 +1314,7 @@ localizedStrings["Press %s to start running the audit."] = "Press %s to start ru
 localizedStrings["Press %s to stop running."] = "Press %s to stop running.";
 localizedStrings["Pressed"] = "Pressed";
-localizedStrings["Click to pretty print"] = "Click to pretty print";
+localizedStrings["Pretty print"] = "Pretty print";
```
this seems out of topic.

jernoble replied:
This is just a result of running `update-webkit-localizable-strings`; it likely caught string changes or sorted differently.

jyavenard commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -3168,6 +3168,28 @@ Inspector::Protocol::ErrorStringOr<void> InspectorDOMAgent::setAllowEditingUserA
+    case VideoProjectionMetadataKind::Parametric:
+        return Inspector::Protocol::DOM::VideoProjectionMetadataKind::Parametric;
+    case VideoProjectionMetadataKind::Pyramid:
+        return Inspector::Protocol::DOM::VideoProjectionMetadataKind::Pyramid;
```
What is "Pyramid" used for? 

as it is in the code, this value can never be set either. Do we need it ?

jernoble replied:
Not used by our port yet, but it is a common projection kind. Think of it as future-proofing.

jyavenard reviewed (approved):
LGTM, relevant issues in test failures, wrong thread assertion it seems.

jernoble committed:
Add logging and Web Inspector support for spatial and projection metadata
rdar://150615061
https://bugs.webkit.org/show_bug.cgi?id=292507

Reviewed by Jean-Yves Avenard.

Add an enumerated list of projection types, and pass that metadata as well
as spatialVideoMetadata through to the Web Inspector, as well as the UIProcess.

Drive-by fixes:
- Add a change observer to VideoTrackConfiguration to allow clients to be notified
  of changes to the configuration without requiring a Client superclass.
- Make AVTrackPrivateAVFObjCImpl reference-counted.
- Compare AVAssetTrack.mediaType and AVMediaSelectionOption.mediaType with -isEqualToString:
  rather than pointer equality.

* Source/JavaScriptCore/inspector/protocol/DOM.json:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Logger.h:
(WTF::LogArgument<std::optional<T>>::toString):
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
(WTF::LogArgument<std::optional<T>>::toString): Deleted.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::audioTrackConfigurationChanged):
(WebCore::HTMLMediaElement::videoTrackConfigurationChanged):
(WebCore::HTMLMediaElement::addAudioTrack):
(WebCore::HTMLMediaElement::addVideoTrack):
(WebCore::HTMLMediaElement::removeAudioTrack):
(WebCore::HTMLMediaElement::removeVideoTrack):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::descriptionForTrack):
(WebCore::MediaElementSession::description const):
* Source/WebCore/html/MediaElementSession.h:
* Source/WebCore/html/track/AudioTrack.cpp:
(WebCore::AudioTrack::configurationChanged):
* Source/WebCore/html/track/AudioTrackClient.h:
(WebCore::AudioTrackClient::audioTrackConfigurationChanged):
* Source/WebCore/html/track/VideoTrack.cpp:
(WebCore::VideoTrack::configurationChanged):
* Source/WebCore/html/track/VideoTrackClient.h:
(WebCore::VideoTrackClient::videoTrackConfigurationChanged):
* Source/WebCore/html/track/VideoTrackConfiguration.cpp:
(WebCore::VideoTrackConfiguration::setState):
(WebCore::VideoTrackConfiguration::setCodec):
(WebCore::VideoTrackConfiguration::setWidth):
(WebCore::VideoTrackConfiguration::setHeight):
(WebCore::VideoTrackConfiguration::setColorSpace):
(WebCore::VideoTrackConfiguration::setFramerate):
(WebCore::VideoTrackConfiguration::setBitrate):
(WebCore::VideoTrackConfiguration::setSpatialVideoMetadata):
(WebCore::VideoTrackConfiguration::setVideoProjectionMetadata):
(WebCore::VideoTrackConfiguration::notifyObservers):
(WebCore::VideoTrackConfiguration::toJSON const):
* Source/WebCore/html/track/VideoTrackConfiguration.h:
(WebCore::VideoTrackConfiguration::videoProjectionMetadata const):
(WebCore::VideoTrackConfiguration::setState): Deleted.
(WebCore::VideoTrackConfiguration::setCodec): Deleted.
(WebCore::VideoTrackConfiguration::setWidth): Deleted.
(WebCore::VideoTrackConfiguration::setHeight): Deleted.
(WebCore::VideoTrackConfiguration::setColorSpace): Deleted.
(WebCore::VideoTrackConfiguration::setFramerate): Deleted.
(WebCore::VideoTrackConfiguration::setBitrate): Deleted.
(WebCore::VideoTrackConfiguration::setSpatialVideoMetadata): Deleted.
(WebCore::VideoTrackConfiguration::isImmersiveVideo const): Deleted.
(WebCore::VideoTrackConfiguration::setIsImmersiveVideo): Deleted.
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::videoProjectionMetadataKind):
(WebCore::InspectorDOMAgent::getMediaStats):
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
(WebCore::PlaybackSessionModelClient::supportsLinearMediaPlayerChanged):
(WebCore::PlaybackSessionModelClient::spatialVideoMetadataChanged):
(WebCore::PlaybackSessionModelClient::videoProjectionMetadataChanged):
(WebCore::PlaybackSessionModelClient::isImmersiveVideoChanged): Deleted.
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
(WebCore::PlaybackSessionModelMediaElement::videoTrackConfigurationChanged):
(WebCore::PlaybackSessionModelMediaElement::maybeUpdateVideoMetadata):
(WebCore::PlaybackSessionModelMediaElement::seekableRanges const):
(WebCore::PlaybackSessionModelMediaElement::isInWindowFullscreenActive const):
(WebCore::PlaybackSessionModelMediaElement::logIdentifier const):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::convertEnumerationToString):
(WebCore::convertVideoProjectionMetadataToString):
* Source/WebCore/platform/graphics/PlatformVideoTrackConfiguration.h:
* Source/WebCore/platform/graphics/VideoProjectionMetadata.h: Copied from Source/WebCore/html/track/VideoTrackClient.h.
(WTF::LogArgument<WebCore::VideoProjectionMetadata>::toString):
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.h:
* Source/WebCore/platform/graphics/avfoundation/AVTrackPrivateAVFObjCImpl.mm:
(WebCore::AVTrackPrivateAVFObjCImpl::initializeAssetTrack):
(WebCore::AVTrackPrivateAVFObjCImpl::initializationCompleted):
(WebCore::AVTrackPrivateAVFObjCImpl::readyState const):
(WebCore::AVTrackPrivateAVFObjCImpl::videoTrackConfiguration const):
(WebCore::AVTrackPrivateAVFObjCImpl::spatialVideoMetadata const):
(WebCore::AVTrackPrivateAVFObjCImpl::videoProjectionMetadata const):
(WebCore::AVTrackPrivateAVFObjCImpl::isImmersiveVideo const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp:
(WebCore::spatialVideoMetadataFromFormatDescription):
(WebCore::videoProjectionMetadataFromFormatDescription):
(WebCore::videoMetadataFromFormatDescription): Deleted.
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.h:
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateAVFObjC.mm:
(WebCore::AudioTrackPrivateAVFObjC::AudioTrackPrivateAVFObjC):
(WebCore::AudioTrackPrivateAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateMediaSourceAVFObjC.cpp:
(WebCore::AudioTrackPrivateMediaSourceAVFObjC::AudioTrackPrivateMediaSourceAVFObjC):
* Source/WebCore/platform/graphics/avfoundation/objc/AudioTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/TextTrackPrivateMediaSourceAVFObjC.cpp:
(WebCore::m_impl):
* Source/WebCore/platform/graphics/avfoundation/objc/TextTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateAVFObjC.cpp:
(WebCore::VideoTrackPrivateAVFObjC::VideoTrackPrivateAVFObjC):
(WebCore::VideoTrackPrivateAVFObjC::resetPropertiesFromTrack):
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoTrackPrivateMediaSourceAVFObjC.mm:
(WebCore::VideoTrackPrivateMediaSourceAVFObjC::VideoTrackPrivateMediaSourceAVFObjC):
* Source/WebCore/platform/graphics/cocoa/NullVideoPresentationInterface.h:
* Source/WebCore/platform/graphics/cocoa/VideoTrackPrivateWebM.cpp:
(WebCore::VideoTrackPrivateWebM::updateConfiguration):
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js:
(WI.MediaDetailsSidebarPanel):
(WI.MediaDetailsSidebarPanel.prototype.layout):
(WI.MediaDetailsSidebarPanel.prototype.initialLayout):
(WI.MediaDetailsSidebarPanel.prototype.localizedVideoProjectionMetadataKindString):
(WI.MediaDetailsSidebarPanel.prototype.setVideo):
* Source/WebKit/Platform/IPC/ArgumentCoders.cpp:
(IPC::ArgumentCoder<JSON::Value>::encode):
(IPC::ArgumentCoder<JSON::Value>::decode):
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.h:
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.mm:
(WebKit::PlaybackSessionInterfaceLMK::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionInterfaceLMK::isImmersiveVideoChanged): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionModelContext::spatialVideoMetadataChanged):
(WebKit::PlaybackSessionModelContext::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionManagerProxy::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionModelContext::isImmersiveVideoChanged): Deleted.
(WebKit::PlaybackSessionManagerProxy::isImmersiveVideoChanged): Deleted.
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionInterfaceContext::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionManager::videoProjectionMetadataChanged):
(WebKit::PlaybackSessionInterfaceContext::isImmersiveVideoChanged): Deleted.
(WebKit::PlaybackSessionManager::isImmersiveVideoChanged): Deleted.

Canonical link: https://commits.webkit.org/294733@main
